File: study-kotlin/java-to-kotlin/kotlin/src/test/kotlin/lec12_object/javacode/Person.java
package lec12_object.javacode;

public class Person {
	private static final int MIN_AGE = 1;

	private final String name;
	private final int age;

	public Person(final String name, final int age) {
		this.name = name;
		this.age = age;
	}


	private static Person newBaby(String name){
		return new Person(name, MIN_AGE);
	}
}

File: study-kotlin/java-to-kotlin/kotlin/src/test/kotlin/lec12_object/code/Person.kt
package lec12_object.code

class Person private constructor(
    var name: String,
    var age: Int
) {

    companion object Factory : Log {
        const val MIN_AGE = 1 // const -> 변수가 compile 시에 할당 된다, 기본 타입과 String 에만 붙일 수 있다
        fun newBaby(name: String): Person {
            return Person(name, MIN_AGE)
        }

        override fun log() {
            println("hello companion object")
        }
    }
}